src/solvers/dgsem_tree/dg_2d.jl

```diff
@@ -974,21 +974,33 @@ function calc_surface_integral!(du, u, mesh::Union{TreeMesh{2}, StructuredMesh{2
   factor_1 = boundary_interpolation[1,          1]
   factor_2 = boundary_interpolation[nnodes(dg), 2]
   @threaded for element in eachelement(dg, cache)
-    for l in eachnode(dg)
+    # surface at -x
+    @turbo for l in eachnode(dg)
```

Comment:
Did we discuss having `@turbo` optimizations in our default code? I don't recall anything, but I've been in information overload for the past weeks, so MMMV ;-)

Reply:
Same question applies to `dgsem_tree/dg_3d.jl`.

Reply:
We do use `@turbo` in standard code, e.g., 
https://github.com/trixi-framework/Trixi.jl/blob/5fd8a0711700f9e678d627a9a79a73a75373855c/src/solvers/dgsem_structured/containers_2d.jl#L71

Reply:
IIRC, this was a special case since the container reinitialization was super slow for AMR. We have never used it in "numerics" functions so far (only for interpolation/`multiply_dimensionwise` if I am not mistaken).

Thus, for the sake of making this mergeable, I'd suggest to not apply these optimizations in these very select functions but rather leave this for another day. Before we start using `@turbo`, `@avx`, `@simd` etc. in regular kernels, I think we should first have a discussion about it in general, and then also explain it in the docs what it does, its caveats etc.

Reply:
Tracked in #999

Reply:
Okay, let's continue this discussion elsewhere. I removed the changes of this surface stuff.

Reply:
👍